java/org/apache/catalina/valves/ExtendedAccessLogValve.java

```diff
@@ -298,12 +298,22 @@ public CookieElement(String name) {
         @Override
         public void addElement(CharArrayWriter buf, Date date, Request request,
                 Response response, long time) {
+            StringBuilder value = new StringBuilder();
+            boolean first = true;
             Cookie[] c = request.getCookies();
             for (int i = 0; c != null && i < c.length; i++) {
                 if (name.equals(c[i].getName())) {
-                    buf.append(wrap(c[i].getValue()));
+                    if (first) {
+                        first = false;
+                    } else {
+                        value.append(",");
+                    }
+                    value.append(wrap(c[i].getValue()));
```

Comment:
Need to wrap the output rather than individual values.

Reply:
Fixed